Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/jee/jsf/UserAuthorizationListener.java

```diff
@@ -154,4 +154,27 @@ private boolean checkConfiguredDisplay(String currentPage) {
         return true;
     }
 
+    /**
+     * @param event 
+     * 
+     *when a transition happens from status.xhtml to properties.xhtml the csrf token will be validated for status.xhtml 
+     *and afterPhase method will be again executed and it will try to verify the token for properties.xhtml.The validation is 
+     *now performed by checking if token is null and ignore the transitioned page. Below "if" method can be re-written in 
+     *a way so that if token is null and the phase of transition can be checked it will be safer to ignore that validation.
+     */
+    public void validateCSRFToken(PhaseEvent event) {
+        String csrfToken = event.getFacesContext().getExternalContext().getRequestParameterMap().get("csrfToken");
+        if (csrfToken != null) {
+            LOG.debug("csrfToken from http Request parameter is not null");
+            if (csrfToken.equals(event.getFacesContext().getExternalContext().getSessionMap().get("salt"))) {
+                LOG.debug("CSRF Token successfully validated");
+            } else {
+                LOG.debug("CSRF Token validation Failed");
```

Comment:
This is an error condition that kicks the user out, we need an error message that indicates that the session is being invalidated (but doesn't go into details about csrf tokens).  Original debug statement can supplement the new one if you think it will still be useful.